### ISSUE TITLE
latest 1.34 release fail to compile on Mageia Linux

### ISSUE THREAD

**Issue by david-geiger** [rzvncj/xCHM#26]
Hi,

I tried to compile latest 1.34 release on Mageia Linux Cauldron but it fails to build with:

chmframe.cpp:855:21: error: call of overloaded 'AddTool(<unnamed enum>, const wxString&, wxBitmap, const wxString&)' is ambiguous

Attached full build log:
[build.x86_64.0.20230205060704.log](https://github.com/rzvncj/xCHM/files/10610344/build.x86_64.0.20230205060704.log)

**Comment by rzvncj** [comment]
Could you check to see if the commit I've pushed fixes you problem, please?

**Comment by rzvncj** [comment]
I can't reproduce it at all on my Arch Linux - even by using all of your `CFLAGS`, `CXXFLAGS`, `LDFLAGS`, etc.
I also have wxWidgets 3.2.1 here.

If the commit above does not fix it, would it be possible for you to try on an up-to-date Arch Linux and reproduce it there?

**Comment by rzvncj** [comment]
Nevermind, I've figured it out. Your wxWidgets, although appearing to be version 3.2.1, has enabled backwards compatibility with 2.8, and that's why it's hitting this overload (in /usr/include/wx-3.2/wx/tbarbase.h:503):
```c++
498 #if WXWIN_COMPATIBILITY_2_8
499     // the old versions of the various methods kept for compatibility
500     // don't use in the new code!
501     // --------------------------------------------------------------
502     wxDEPRECATED_INLINE(
503     wxToolBarToolBase *AddTool(int toolid,
504                                const wxBitmap& bitmap,
505                                const wxBitmap& bmpDisabled,
506                                bool toggle = false,
507                                wxObject *clientData = NULL,
508                                const wxString& shortHelpString = wxEmptyString,
509                                const wxString& longHelpString = wxEmptyString)
510     ,
511         return AddTool(toolid, wxEmptyString,
512                        bitmap, bmpDisabled,
513                        toggle ? wxITEM_CHECK : wxITEM_NORMAL,
514                        shortHelpString, longHelpString, clientData);
515     )
```
The cleanest solution, IMHO, is for you to build the Mageia wxWidgets package in a way that removes backwards compatibility with the (seriously old) 2.8 version, and then your build problem goes away.

(Also, the previous patch will not fix this issue, so no need to check if the problem's now solved with it - it isn't).

**Comment by rzvncj** [comment]
You'll also note that in the current stable documentation for wxWidgets, the offending overload is not listed at all: https://docs.wxwidgets.org/3.2/classwx_tool_bar.html#a5987506e585f03c294c55799beca934a, so it is not meant to be used (or even taken into consideration for overload purposes) for builds of new software.

**Comment by rzvncj** [comment]
I've just released 1.35, it should build even with your current wxWidgets.

**Comment by david-geiger** [comment]
Confirmed it is fixed with new 1.35 release, thank you.

**Comment by rzvncj** [comment]
> Confirmed it is fixed with new 1.35 release, thank you.

Alright, no problem. Thanks for reporting.
